apps/wallet/src/ui/app/components/accounts/EditableAccountName.tsx

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useZodForm } from '@mysten/core';
+import { type ComponentProps, forwardRef } from 'react';
+import toast from 'react-hot-toast';
+import { z } from 'zod';
+import { useAccounts } from '../../hooks/useAccounts';
+import { useBackgroundClient } from '../../hooks/useBackgroundClient';
+import { Form } from '../../shared/forms/Form';
+
+type InputProps = Omit<ComponentProps<'input'>, 'className'>;
+
+export const Input = forwardRef<HTMLInputElement, InputProps>((props, forwardedRef) => (
+	<input
+		className="transition peer items-center border-none outline-none bg-transparent hover:text-hero rounded-sm text-pBody text-steel-darker font-semibold p-0 focus:bg-transparent"
+		ref={forwardedRef}
+		{...props}
+		id="current-address-nickname-edit"
+	/>
+));
+
+const formSchema = z.object({
+	nickname: z.string().trim(),
+});
+
+export function EditableAccountName({ accountID, name }: { accountID: string; name: string }) {
+	const backgroundClient = useBackgroundClient();
+	const { data: accounts } = useAccounts();
+	const account = accounts?.find((account) => account.id === accountID);
```

Comment:
could possibly pass in account as a prop instead of accountID to avoid this lookup